cmd/internal/kube/inject.go

```diff
@@ -184,6 +221,43 @@ func createSidecar(projectName string) v1.Container {
 	return sidecar
 }
 
+func createPostmanSidecar(postmanCollectionID string, postmanEnvironment string) v1.Container {
+	sidecar := v1.Container{
+		Name:  "akita",
+		Image: "akitasoftware/cli:latest",
+		Env: []v1.EnvVar{
+			{
+				Name: "AKITA_POSTMAN_API_KEY",
+				ValueFrom: &v1.EnvVarSource{
+					SecretKeyRef: &v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "akita-postman-secrets",
+						},
+						Key: "postman-api-key",
+					},
+				},
+			},
+		},
+		Lifecycle: &v1.Lifecycle{
+			PreStop: &v1.LifecycleHandler{
+				Exec: &v1.ExecAction{
+					Command: []string{
+						"/bin/sh",
+						"-c",
+						"AKITA_PID=$(pgrep akita) && kill -2 $AKITA_PID && tail -f /proc/$AKITA_PID/fd/1",
+					},
+				},
+			},
+		},
+		Args: []string{"apidump", "--collection", postmanCollectionID, "--environment", postmanEnvironment},
```

Comment:
As things stand currently in the akita-cli side we also need to specify `--domain staging.akita.software` so that the CLI talks to the correct API endpoint.

I do not like this answer, so the alternative is to derive the domain (staging.akita.software) from the choice of postman environment in akita-cli.